build_tools/waf_dynamo.py

```diff
@@ -1206,7 +1210,7 @@ def linux_link_flags(self):
 @after('apply_obj_vars')
 def js_web_link_flags(self):
     platform = self.env['PLATFORM']
-    if 'web' in platform:
+    if 'web' in platform and 'test' in self.features:
```

Comment:
Noticed that we added some code for headless mode, into all out executables.
This code is only for the tests.

We _do_ have one for the engine, but we don't seem to use it?